projects/extension/build.py

```diff
@@ -583,7 +584,33 @@ def build() -> None:
     build_sql()
 
 
+def error_if_pre_release() -> None:
+    # Note: released versions always have the output sql file commited into the repository.
+    output_file = output_sql_file()
+    command = (
+        "just ext build-install"
+        if "ROOT_JUSTFILE" in os.environ
+        else "just build-install"
+        if "PROJECT_JUSTFILE" in os.environ
+        else "python3 build.py build-install"
+    )
+    if not Path(output_file).exists():
+        print(
+            textwrap.dedent(f"""
+                WARNING: You're trying to install a pre-release version of pgai.
+                This is not supported, and there is no upgrade path.
+
+                Instead, install an official release from https://github.com/timescale/pgai/releases.
+
+                If you are certain that you want to install a pre-release version, run:
+                    `{command}`
+            """)
+        )
+        exit(1)
+
+
 def install() -> None:
+    error_if_pre_release()
```

Comment:
Should this be called here or in install_sql()?

Reply:
i.e. do we only want to catch this if they do `just install` or also if they do `just install-sql` as well?

Reply:
I put it here specifically (at the top of this function), and not in `install_sql` so that it aborts early (and not after doing a bunch of pip installs etc.). I can put it in `install_sql` too, although I'm not too concerned about that path because we don't really document it anywhere.

Reply:
makes sense.